src/components/demo-tabs.tsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import { StaticQuery, graphql } from 'gatsby';
+import { ContentPageData } from '../types';
+import AuthorsList from './authors-list';
+import EditLink from './edit-link';
+import { ParsedContent } from '../util/ParsedContent';
+
+const query = graphql`
+  query DemoTabsContent {
+    content: markdownRemark(
+      frontmatter: {
+        title: { eq: "Installing via Package Manager" }
+        section: { eq: "Content" }
+      }
+    ) {
+      id
+      html
+      tableOfContents
+      frontmatter {
+        title
+        description
+      }
+      fields {
+        slug
+        authors
+      }
+      parent {
+        ... on File {
+          relativePath
+        }
+      }
+    }
+  }
+`;
+
+export interface Props {
+  parsedContent?: ParsedContent;
+}
+
+class DemoTabs extends React.Component<Props> {
+  public parsedContent: ParsedContent;
+
+  public parsedFragment: DocumentFragment | null;
+
+  public articleRef: React.RefObject<HTMLElement>;
+
+  public selectRef: React.RefObject<HTMLSelectElement>;
+
+  public optionMap: WeakMap<HTMLOptionElement, HTMLElement>;
+
+  public options: HTMLOptionElement[];
+
+  public sections: HTMLElement[];
+
+  public currentSection: HTMLElement | null;
+
+  public constructor(props: Props) {
+    super(props);
+    this.parsedContent = (props && props.parsedContent) || new ParsedContent();
+    this.articleRef = React.createRef();
+    this.selectRef = React.createRef();
+    this.optionMap = new WeakMap<HTMLOptionElement, HTMLElement>();
+    this.options = [];
+    this.sections = [];
+    this.currentSection = null;
+    this.parsedFragment = null;
+  }
+
+  public componentDidMount() {
+    this.populate();
+  }
+
+  public populate() {
+    if (this.parsedFragment === this.parsedContent.documentFragment) return;
+
+    /* eslint-disable */
```

Comment:
Let's make sure we re-enable this and run `yarn lint:fix` before merging :)

Reply:
Yup… but I will be honest, I have issues with some rules that I am not sure even therapy will resolve 🤣

Reply:
To clarify, I am slowly wrapping my head around some rules with wet feat, but some I really feel like I missed a few chapters in the metaphorical book of logic:

```js
class X {
  static manipulateInstance(instance: X) {
    x.state.something = true;
  }
}
```

💭 Very neatly encapsulated code and a pattern that cannot cut the wrong way, imho.

> ‼️ Ah Oh! Assigning to a property of an argument — eslint

🗯 Did you even read know it was a class?!

```js
/* eslint-disable */
…
/* eslint-enable */
```

💭 Very neatly encapsulated code and a pattern that cannot cut the wrong way, imho.